tests/drivers/candev/main.c

```diff
@@ -220,36 +226,67 @@ if (IS_ACTIVE(CONFIG_USE_LOOPBACK_MODE)) {
     canopt_state_t mode = CANOPT_STATE_LOOPBACK;
     candev->driver->set(candev, CANOPT_STATE, &mode, sizeof(mode));
 }
-
+#if defined(MODULE_MCP2515)
     if (IS_ACTIVE(MCP2515_RECV_FILTER_EN)) {
+#endif
         /* CAN filters examples */
         struct can_filter filter[4];
         filter[0].can_mask = 0x7FF;
         filter[0].can_id = 0x001;
 #if defined(MODULE_MCP2515)
         filter[0].target_mailbox = 0;   /* messages with CAN ID 0x001 will be received in mailbox 0 */
+#elif defined(MCU_SAMD5X)
+        filter[0].can_filter_type = CAN_FILTER_TYPE_CLASSIC;
+        filter[0].can_filter_conf = CAN_FILTER_RX_FIFO_0; /* meassages with CAN ID 0x001 will be received in Rx FIFO 0 */
 #endif
         filter[1].can_mask = 0x7FF;
         filter[1].can_id = 0x002;
 #if defined(MODULE_MCP2515)
         filter[1].target_mailbox = 1;   /* messages with CAN ID 0x002 will be received in mailbox 1 */
+#elif defined(MCU_SAMD5X)
+        filter[1].can_filter_type = CAN_FILTER_TYPE_CLASSIC;
+        filter[1].can_filter_conf = CAN_FILTER_RX_FIFO_1;   /* messages with CAN ID 0x002 will be received in Rx FIFO 1 */
 #endif
         filter[2].can_mask = 0x7FF;
         filter[2].can_id = 0x003;
 #if defined(MODULE_MCP2515)
         filter[2].target_mailbox = 0;   /* messages with CAN ID 0x003 will be received in mailbox 0 */
+#elif defined(MCU_SAMD5X)
+        filter[2].can_filter_type = CAN_FILTER_TYPE_CLASSIC;
+        filter[2].can_filter_conf = CAN_FILTER_RX_REJECT;   /* messages with CAN ID 0x003 will be rejected */
 #endif
         filter[3].can_mask = 0x7FF;
         filter[3].can_id = 0x004;
 #if defined(MODULE_MCP2515)
         filter[3].target_mailbox = 0;   /* this filter won't be applied. Reason is no space found in the first mailbox as it supports only two filters */
+#elif defined(MCU_SAMD5X)
+        filter[3].can_filter_type = CAN_FILTER_TYPE_CLASSIC;
+        filter[3].can_filter_conf = CAN_FILTER_RX_FIFO_1;   /* messages with CAN ID 0x004 will be received in Rx FIFO 1 */
 #endif
         for (uint8_t i = 0; i < 4; i++) {
             candev->driver->set_filter(candev, &filter[i]);
         }
+#if defined(MCU_SAMD5X)
```

Comment:
```suggestion
#if defined(MCU_SAMD5X) && defined(MODULE_PERIPH_CANDEV)
```

we can still use `mcp2515` with SAMD 51